### PR TITLE
ci: build on golang 1.16 and 1.17

### PR DESCRIPTION
**Pull request by leseb**
The CI now runs the build test against Golang 1.15 and 1.16

Signed-off-by: Sébastien Han <seb@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by leseb** [comment]
waiting for https://github.com/portworx/sched-ops/pull/249

**Comment by leseb** [comment]
> waiting for [portworx/sched-ops#249](https://github.com/portworx/sched-ops/pull/249)

Abandoning this since 1.17 is out and the PR has not received any interest.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -57,24 +57,32 @@ jobs:
...
       uses: docker/setup-qemu-action@master
       with:
         platforms: all
 
-    - name: build.all rook
-      run: tests/scripts/github-action-helper.sh build_rook_all
+    - name: build.all rook with go ${{ matrix.go-version }}
+      run: |
+        if [[ "${{ matrix.go-version }}" == "1.17" ]]; then
+          sed -i 's/go 1.16/go 1.17/g' go.mod
```
Comment (travisn) [inline .github/workflows/build.yml:83]: Why does this need to change? I believe we want to keep it to 1.16 so the go modules will be pulled in according to the least supported version.

Comment (leseb) [reply .github/workflows/build.yml]: Good point that's not needed for now., let's just test the build on 1.17 only without pulling deps.

**Comment by leseb** [comment]
We should resurrect this again once https://github.com/kubernetes/gengo/pull/210 merges.

**Pull request merged**